.github/workflows/format-xml.yml

```diff
@@ -0,0 +1,28 @@
+name: Format XML File
+
+on:
+  pull_request:
+    types: [closed]
+    paths:
+      - 'LiveSplit.AutoSplitters.xml'
+
+jobs:
+  format:
+    if: github.event.pull_request.merged == true
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Format .xml file with yq
+      uses: mikefarah/yq@v4.35.2
+      with:
+        cmd: yq -i -ox -I4 '.' LiveSplit.AutoSplitters.xml
+
+    - name: Commit changes
+      uses: EndBug/add-and-commit@v9.1.3
+      with:
+        author_name: github-actions
+        author_email: 41898282+github-actions[bot]@users.noreply.github.com
+        message: Automatic formatting of XML
```

Comment:
Could you link an example of a PR where thiw workflow ran and made a commit? Which branch does this push to? I'm not sure how this would have permission to push to `master`

Reply:
I've created a sample repository at [just-ero/format-xml-on-merged-pr-sample](https://github.com/just-ero/format-xml-on-merged-pr-sample).

https://github.com/just-ero/format-xml-on-merged-pr-sample/pull/1 introduces changes to a sample XML file which need to be formatted. This [runs the action](https://github.com/just-ero/format-xml-on-merged-pr-sample/actions/runs/6348126079), which created and pushed https://github.com/just-ero/format-xml-on-merged-pr-sample/commit/4f1af85c4c46a9caa43774b9be026dc1c98afc8f to `main`.

https://github.com/just-ero/format-xml-on-merged-pr-sample/pull/2 introduces changes to the same XML file which need no formatting. This still [runs the action](https://github.com/just-ero/format-xml-on-merged-pr-sample/actions/runs/6348157668), but since no changes to any files were made, nothing is staged and therefore nothing is committed.